File: Demos/CSharp/16. Status CFDI/SslValidator.cs
using System;
using System.Net;
using System.Net.Security;
using System.Security.Cryptography.X509Certificates;

namespace HyperSoft.Ejemplo.StatusCfdi
{
  internal sealed class SslValidator
  {
    #region Vars

    private RemoteCertificateValidationCallback orgCallback;
    private bool expect100Continue;
    private SecurityProtocolType securityProtocol;

    #endregion

    #region Methods

    internal void OverrideValidation()
    {
      this.expect100Continue = ServicePointManager.Expect100Continue;
      this.orgCallback = ServicePointManager.ServerCertificateValidationCallback;
      this.securityProtocol = ServicePointManager.SecurityProtocol;

      ServicePointManager.ServerCertificateValidationCallback = this.OnValidateCertificate;
      ServicePointManager.Expect100Continue = true;

      try
      {
        const SecurityProtocolType Tls11 = (SecurityProtocolType)768;
        const SecurityProtocolType Tls12 = (SecurityProtocolType)3072;

        ServicePointManager.SecurityProtocol = SecurityProtocolType.Ssl3 | SecurityProtocolType.Tls | Tls11 | Tls12;
      }
      catch (Exception)
      {
        // 
      }
    }

    internal void RestoreValidation()
    {
      ServicePointManager.SecurityProtocol = this.securityProtocol;
      ServicePointManager.ServerCertificateValidationCallback = this.orgCallback;
      ServicePointManager.Expect100Continue = this.expect100Continue;
    }

    private bool OnValidateCertificate(object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)
    {
      return true;
    }

    #endregion
  }
}

File: Demos/Visual Basic .Net/16. Status CFDI/SslValidator.vb
Imports System
Imports System.Net
Imports System.Net.Security
Imports System.Security.Cryptography.X509Certificates

Friend NotInheritable Class SslValidator
	#Region "Vars"

	Private _orgCallback As RemoteCertificateValidationCallback
	Private _expect100Continue As Boolean
	Private _securityProtocol As SecurityProtocolType

	#End Region

	#Region "Methods"

	Friend Sub OverrideValidation()
		Me._expect100Continue = ServicePointManager.Expect100Continue
		Me._orgCallback = ServicePointManager.ServerCertificateValidationCallback
		Me._securityProtocol = ServicePointManager.SecurityProtocol

		'ServicePointManager.ServerCertificateValidationCallback = Me.OnValidateCertificate
		ServicePointManager.Expect100Continue = True

		Try
			Const tls11 As SecurityProtocolType = DirectCast(768, SecurityProtocolType)
			Const tls12 As SecurityProtocolType = DirectCast(3072, SecurityProtocolType)

			ServicePointManager.SecurityProtocol = SecurityProtocolType.Ssl3 Or SecurityProtocolType.Tls Or Tls11 Or Tls12
				
		Catch generatedExceptionName As Exception
		End Try
	End Sub

	Friend Sub RestoreValidation()
		ServicePointManager.SecurityProtocol = Me._securityProtocol
		ServicePointManager.ServerCertificateValidationCallback = Me._orgCallback
		ServicePointManager.Expect100Continue = Me._expect100Continue
	End Sub

  'Private Function OnValidateCertificate(sender As Object, certificate As X509Certificate, chain As X509Chain, sslPolicyErrors As SslPolicyErrors) As Boolean
  '  Return True
  'End Function

#End Region
End Class